nvtabular/io.py

```diff
@@ -331,123 +332,178 @@ def __iter__(self):
             yield from GPUFileIterator(path, **self.kwargs)
 
 
-class Shuffler:
-    """
-    Shuffling the data is an important part of machine learning
-    training. This class is used by Workflow class and shuffles
-    the data after all the pre-processing and feature engineering
-    operators are finished their processing.
+class Writer:
+    def __init__(
+        self,
+        out_dir,
+        num_out_files=30,
+        num_threads=4,
+        output_format="parquet",
+        cats=None,
+        conts=None,
+        labels=None,
+    ):
+        # set variables
+        self.num_out_files = num_out_files
+        self.writer = None
+        if output_format == "binary":
+            self.writer = HugeCTRWriter(out_dir, num_out_files, num_threads, cats, conts, labels)
+        elif output_format == "parquet":
+            self.writer = ParquetFWriter(out_dir, num_out_files, num_threads, cats, conts, labels)
```

Comment:
This shouldn't be in the base Writer class - HugeCTRWriter and ParquetFWriter are both subclasses of Writer.